cmd/sdk-server/main.go

```diff
@@ -53,15 +54,17 @@ const (
 	podNamespaceEnv   = "POD_NAMESPACE"
 
 	// Flags (that can also be env vars)
-	localFlag       = "local"
-	fileFlag        = "file"
-	testFlag        = "test"
-	testSdkNameFlag = "sdk-name"
-	addressFlag     = "address"
-	delayFlag       = "delay"
-	timeoutFlag     = "timeout"
-	grpcPortFlag    = "grpc-port"
-	httpPortFlag    = "http-port"
+	localFlag                 = "local"
+	fileFlag                  = "file"
+	testFlag                  = "test"
+	testSdkNameFlag           = "sdk-name"
+	kubeconfigFlag            = "kubeconfig"
+	noGracefulTerminationFlag = "no-graceful-termination"
```

Comment:
🤔 do we need this? if it's a fully local experience, is it better to not shutdown until the GameServer moves to `Shutdown` state? (it can also be ctrl+c'd I think?)

Reply:
>  if it's a fully local experience, is it better to not shutdown until the GameServer moves to Shutdown state?

If you're developing locally (including running your server in a debugger, setting breakpoints, exiting, re-running, hitting exceptions, crashing, etc) then you probably won't be completing the normal server lifetime. It would be really clunky to _require_ a full server lifetime (progressing through all states) in order to do the most basic debugging -- especially in the context of quickly iterating and you want to run your game server to just try something real quick.

> (it can also be ctrl+c'd I think?)

Unfortunately, no. The open connections checking server state will persist the binary. Spamming `^C` will not work, requiring a `kill -9` to get it to exit.

Preventing the process from exiting was the point of "graceful termination". See https://github.com/googleforgames/agones/pull/2205. As written in that PR, it prevents termination unless `SIGKILL` is received.

In case you want to try this yourself: do note that running the SDK Server then pressing `^C` will properly exit. However, after any state updates, like a call to `agonesSdk.ReadyAsync()` will then get state monitoring to keep the processes alive. However, passing `--no-graceful-termination` will always allow for exiting with `^C` at any point.

Reply:
That's fair, and makes sense. My only thought here then would be to make the flag a positive name with a default of `true` rather than a negative with a default of `false`.

So `--graceful-termination=false` would probably be easier to reason with. WDYT?

Reply:
I think `--blah=false` (with `--blah=true` as default) vs `--no-blah` is the same to me. I've definitely seen various binaries use either approach. Happy to change this around, if that's your preference.

Reply:
Updated. Renamed to `--graceful-termination` and flipped logic.